src/script/descriptor.cpp

```diff
@@ -1509,6 +1509,10 @@ std::vector<std::unique_ptr<PubkeyProvider>> ParsePubkeyInner(uint32_t key_exp_i
         error = "No key provided";
         return {};
     }
+    if (IsSpace(str.front()) || IsSpace(str.back())) {
```

Comment:
Interestingly this line is correct only because we check `IsHex` before `ParseHex` below, as otherwise we would need to check for any space character across the string since `ParseHex` would skip those even if they happen in the middle.

Checking for a space character at any position would also improve the error message for instance when parsing `pk(03 a34b...)`. But no need to retouch.

Reply:
Ok.